src/main/java/io/cdap/plugin/kudu/KuduSourceConfig.java

```diff
@@ -16,67 +16,152 @@
+  private String optOperationTimeout;
+
+  public KuduSourceConfig(String referenceName, String optMasterAddresses, String optTableName, String optSchema,
+                          @Nullable String optColumnProjection, @Nullable String optOperationTimeout) {
+    this.referenceName = referenceName;
+    this.optMasterAddresses = optMasterAddresses;
+    this.optTableName = optTableName;
+    this.optSchema = optSchema;
+    this.optColumnProjection = optColumnProjection;
+    this.optOperationTimeout = optOperationTimeout;
+  }
+
+  private KuduSourceConfig(Builder builder) {
+    referenceName = builder.referenceName;
+    optMasterAddresses = builder.optMasterAddresses;
+    optTableName = builder.optTableName;
+    optSchema = builder.optSchema;
+    optColumnProjection = builder.optColumnProjection;
+    optOperationTimeout = builder.optOperationTimeout;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(KuduSourceConfig copy) {
+    return builder()
+      .setReferenceName(copy.referenceName)
+      .setOptMasterAddresses(copy.optMasterAddresses)
+      .setOptTableName(copy.optTableName)
+      .setOptSchema(copy.optSchema)
+      .setOptColumnProjection(copy.optColumnProjection)
+      .setOptOperationTimeout(copy.optOperationTimeout);
+  }
+
+  public String getReferenceName() {
+    return referenceName;
+  }
+
+  public String getOptMasterAddresses() {
+    return optMasterAddresses;
+  }
+
+  public String getOptTableName() {
+    return optTableName;
+  }
+
+  public String getOptSchema() {
+    return optSchema;
+  }
+
+  @Nullable
+  public String getOptColumnProjection() {
+    return optColumnProjection;
+  }
+
+  @Nullable
+  public String getOptOperationTimeout() {
+    return optOperationTimeout;
+  }
 
   /**
    * Validates the configuration fields.
    *
    * @throws IllegalArgumentException throw when there is issue with the configuration.
    */
-  public void validate() throws IllegalArgumentException {
-    if (!containsMacro("master")) {
-      Preconditions.checkArgument(!Strings.isNullOrEmpty(optMasterAddresses),
-                                  "Kudu Master Server address list is empty.");
+  public void validate(FailureCollector failureCollector) throws IllegalArgumentException {
+    if (!containsMacro(MASTER)){
+      if (Strings.isNullOrEmpty(optMasterAddresses)) {
+        failureCollector.addFailure("Kudu Master Server address list is empty.",
+                                    "Provide Kudu Master Server address list.")
+          .withConfigProperty(MASTER);
+      } else {
+        Arrays.stream(optMasterAddresses.split(",")).forEach(masterAddress -> {
+          if (masterAddress.split(":").length != 2){
+            failureCollector.addFailure(String.format("Master Address '%s' has incorrect format.", masterAddress),
+                                        "Provide Master Address in format - <hostname>:<port>")
```

Comment:
`<hostname1>:<port1>,<hostname2>:<port2>,...` ?

Reply:
`<hostname1>:<port1>,<hostname2>:<port2>,... `correct format
we check that each master address should be in format `<hostname1>:<port1>`